### DISCUSSION TITLE
Питання / Questions

### DISCUSSION THREAD

**Discussion by DmytroKorniienko** [DmytroKorniienko/FireLamp_EmbUI#103] [Q&A] [answered]
Якщо виникли якись питання стосовно проекту - можете спитати прямо тут, мова спілкування англійська чи українська.

**Comment by Edgar-electro** [comment]
Good day. Can you help me please. There are several problems. I assembled a Guyver lamp with color music. but it does not work correctly. I will send you a video. You can see how it works. Thanks in advance.https://youtu.be/E4nnkEL7dSk  Doesn't match the music at all.

**Reply by kostyamat** [reply]
Hello. 
Judging by the video, there are two assumptions. First: You have not calibrated the microphone.
Second: The microphone is disabled globally, the button is at the top right, there are additional switches. If the microphone is muted, then "Frequency Analyzer" shows a random pattern.
To calibrate the microphone, enter the settings, in the "microphone" item, turn it on and click "Calibration". This process should be done at a normal noise level in the room (city noise outside the window, etc.), do not make additional noise during the process, turn off all sources of additional sounds. The calibration process takes some time.

**Comment by Edgar-electro** [comment]
Good evening.  I did everything Normally, the connection settings calibrated the microphone.  It still doesn't work correctly...We can say FFT does not do its job.I also use a microphone on other projects, the microphone is also working....It is clear that the problem is software.....Thank you very much.  Too bad they couldn't do something.

**Reply by DmytroKorniienko** [reply]
Which version of firmware are you using? I'm shared a quick mic test at esp32 (but this don't matter) where I didn't see any issues: https://vimeo.com/manage/videos/736907340

**Comment by Edgar-electro** [comment]
https://youtu.be/JB-iBaSUxyo

**Comment by Edgar-electro** [comment]
Please see this project.  How everything works correctly, the music clearly reacts, this is another sketch.
I really, really like your project.  I would like to combine these two projects.

**Reply by DmytroKorniienko** [reply]
Which version of our project? Current version 2.7.0 at master/dev most newest - dev branch. And please provide test with tone generator, not just music.

**Reply by Edgar-electro** [reply]
sLava Ukraina

**Comment by Edgar-electro** [comment]
Good, thank you

**Comment by AVL4GITHUB** [comment, marked as answer]
Слава Україні!
Автор живий-здоровий?
Можна щось запитувати?

**Reply by AVL4GITHUB** [reply]
Текуча прошивка в моїй лампі 2.4.0.
Ніяких наворотів я не використовую.
Керую сенсорною кнопкою, або через ВЕБ-морду.
Чи є зміст мені качати-інсталювати-генерувати крайню прошивку?
Вона мені чимось  реально буде корисна?
Чи нові ефекти появляться?
Дякую!

**Reply by St3p40** [reply]
Вітаю
Зміст завантажувати та інсталювати останню прошивку буде.
Точно є багфікси і нові функції.
Ефектів нових в останніх релізах не було, якщо добре пам'ятаю.
Цікаво, звідки ви прочитали про версію, з веб-морди?
Що до ефектів в майбутніх версіях:
Швидше всього, що нових не буде, буде оновлення старих.

**Reply by AVL4GITHUB** [reply]
[user_config.txt](https://github.com/DmytroKorniienko/FireLamp_EmbUI/files/9632523/user_config.txt)
Бажаю здоров'я!
Скомпілюйте, будь ласка, бінарники під мій конфіг.
Зараз нема часу і натхнення вникати чому у мене не компілюється.

Дякую!

**Comment by AVL4GITHUB** [comment]
[user_config.txt](https://github.com/DmytroKorniienko/FireLamp_EmbUI/files/9632523/user_config.txt)
Бажаю здоров'я!
Скомпілюйте, будь ласка, бінарники під мій конфіг.
Зараз нема часу і натхнення вникати чому у мене не компілюється.

Дякую!

**Reply by DmytroKorniienko** [reply]
Який таргет потрібен? Це ESP32 чи ESP8266?

**Reply by AVL4GITHUB** [reply]
ESP8266 потрібен.

**Reply by DmytroKorniienko** [reply]
[esp8266dev.zip](https://github.com/DmytroKorniienko/FireLamp_EmbUI/files/9639679/esp8266dev.zip)

**Reply by AVL4GITHUB** [reply]
Дякую!
Але поки фіаско.
Прошив по повітрю.
Спочатку ФС, потім мейн.
Лампа здохла. Точки доступу нема.
Приніс на роботу лампу, буду реанімовувати.

**Comment by AVL4GITHUB** [comment]
Доповідаю.
Лампу підняв тільки через прошивку по КОМ-порті.
Ще раз перевірив, що по повітрі 270 на 240 не шиється. А 270 на 270 по повітрі - шиється.
Ще, вроді ж і дивився, але значить, не додивився. Кнопка у мене на D3.
Звісно, можна і перепаяти, але я зробив декілька (4 ламп) і не хочеться змінювати мій "стандарт".
Дякую.

**Reply by DmytroKorniienko** [reply]
Пін кнопки можна міняти на інший у налаштуваннях ESP, які до того треба вмикнути.

![Screenshot_2022-09-26-22-07-59-837_com android chrome](https://user-images.githubusercontent.com/26786760/192359806-e4c3d826-4c83-487c-84d5-5ccbd0179969.jpg)

**Reply by AVL4GITHUB** [reply]
Дякую, але яка відповідність номеру піна до назви піна? D3 це який номер?

**Reply by AVL4GITHUB** [reply]
А краще, перегенеруйте, будь ласка, бінарники під кнопку на D3.
Дякую.

**Comment by AVL4GITHUB** [comment]
Вітаю!
Як згенерувати  ФС в VSC?
В Builder.bat ні прошивка ні файлова у мене не генерується.
В VSC прошивка - генерується. А як згенерувати ФС?
Дякую!

**Reply by DmytroKorniienko** [reply]
![image](https://user-images.githubusercontent.com/26786760/194959238-e76ce95e-0f64-4aca-9d33-c5ff0c779a99.png)

**Comment by AVL4GITHUB** [comment]
Слава Україні!
Для тестування блоку живлення для лампи найкраще виставити ліміт по току 4000 і включити ефект "Біла лампа"?
Бо бачу, що часто-густо моя лампа блимає як стробоскоп (перенавантажується  і перезавантажується 2-х амперний блок живлення).

**Reply by DmytroKorniienko** [reply]
4000mA == 4A, що є забагато для 2А блока живлення, таким чином у вас має бути максимальний ліміт в 2000

**Comment by AVL4GITHUB** [comment]
Слава Україні!
Як включити на постійно необхідний ефект з кнопки?
Раніше було так.
Гортаєш вперед чи назад кнопкою до необхідного ефекту.
Вимикаєш лампу.
Вмикаєш лампу  одноразовим дотиком і необхідний ефект постійно працює.
Зараз так не працює.
Вмикається на першому чи якомусь іншому ефекті, а не на тому, на котрому вимкнули.
Що робити?
Дякую.

**Reply by DmytroKorniienko** [reply]
Нічого у керуванні кнопкою не змінювалось, в мене все працює як і працювало. Перевіряйте у себе як кнопку, так і її налаштування. Якщо не знаходите проблеми, то надайте лог вмкання/вимикання.
* Збереження конфігурації йде кожні 30 секунд.

![Screenshot_2022-11-02-22-02-45-665_com android chrome](https://user-images.githubusercontent.com/26786760/199592734-62eb093a-0b5e-40ec-91d1-a94260203dc9.jpg)

**Comment by AVL4GITHUB** [comment]
Слава Україні!
Маю таку неприємність з лампою.
Пропрацювала бездоганно десь днів з 10 в офісі, в робочий час.
Дальше - заглючила. 
А саме, секунд з 30 працює бездоганно, а потім з'являється горизонтальна, широка, зелена смужка і лампа висне, вимикається. 
Лампу можна вручну, чи через вебку увімкнути, але в любому випадку - лампа через 30 секунд здохне.
Через вебку я не вспіваю її перепрошити, бо лампа ребутитиься.
Лампа заклєєна наглухо і доступу до ЮЗБ порту нема.
Якою командою можна швидко попробувати перешити лампу по мережі?
Проблема не в блоці живлення.
Дякую.

**Reply by DmytroKorniienko** [reply]
Яка заповенність ФС? Схоже що при записі конфігу йде перезавантаження. Погано що немає доступу до USB та логів... Для оновлення прошивки можна ввести /update у кінці адреси в браузері, далі завантажити прошивку.

**Comment by AVL4GITHUB** [comment]
Слава Україні!
Маю таку неприємність з лампою.
Пропрацювала бездоганно десь днів з 10 в офісі, в робочий час.
Дальше - заглючила.
А саме, секунд з 30 працює бездоганно, а потім з'являється горизонтальна, широка, зелена смужка і лампа висне, вимикається.
Лампу можна вручну, чи через вебку увімкнути, але в любому випадку - лампа через 30 секунд здохне.
Через вебку я не вспіваю її перепрошити, бо лампа ребутитиься.
Лампа заклєєна наглухо і доступу до ЮЗБ порту нема.
Якою командою можна швидко попробувати перешити лампу по мережі?
Проблема не в блоці живлення.
Дякую.

**Reply by Lumenjer** [reply]
Героям Слава!
Лампа ребутиться навіть якщо виключена (не світить, але можна зайти на веб інтерфейс)?

**Reply by AVL4GITHUB** [reply]
Так, 30 секунд часу є для цього.

**Reply by AVL4GITHUB** [reply]
30 секунд є доступ до вебки.

**Reply by DmytroKorniienko** [reply]
Яка версія прошивки встановлена зараз?

**Comment by dimazn1** [comment]
Hello! Is there an integration or manual for the home assistant to add a lamp? Thank you!

**Reply by DmytroKorniienko** [reply]
Lamp used to standard light integration, you can enum and find it in a mqtt objects:

![Screenshot_2023-02-03-22-58-18-101_com android chrome](https://user-images.githubusercontent.com/26786760/216709493-fd1e5ec2-6c5b-4457-a10b-db0f79f1114e.jpg)
![Screenshot_2023-02-03-22-58-33-094_com android chrome](https://user-images.githubusercontent.com/26786760/216709521-a3010447-a79a-4d4a-a961-cb841e302d52.jpg)

**Comment by zeone** [comment]
Доброго дня, прошив лампу на останню версію (також пробував гілку Dev) але виникла проблема з фронтом, він не рендериться в консолі помилка

> Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'value')
>     at Object.make (embui.js:669:59)
>     at embui.js:770:8

`                        if (!global.menu_id)
                            global.menu_id = global.menu[0].value`
 
 Масив в `menu` приходить пустим
